Userland/Libraries/LibWeb/Bindings/WindowObject.cpp

```diff
@@ -402,6 +403,20 @@ JS_DEFINE_NATIVE_FUNCTION(WindowObject::btoa)
     return JS::js_string(vm, move(encoded));
 }
 
+// https://html.spec.whatwg.org/multipage/window-object.html#dom-length
+JS_DEFINE_NATIVE_FUNCTION(WindowObject::length_getter)
+{
+    auto* impl = TRY(impl_from(vm));
+
+    // 1. If W's browsing context is null, then return 0.
+    auto* this_browsing_context = impl->associated_document().browsing_context();
+    if (!this_browsing_context)
+        return JS::Value(0);
+
+    // 2. Return the number of document-tree child browsing contexts of W's browsing context.
+    return JS::Value(this_browsing_context->document_tree_child_browsing_context_count());
```

Comment:
Neat! Please package this in a standalone function (https://html.spec.whatwg.org/multipage/window-object.html#number-of-document-tree-child-browsing-contexts) and consider fixing these in a second commit:

- https://github.com/SerenityOS/serenity/blob/987927a596cb773aee0069666d650e3f26f9f3ac/Userland/Libraries/LibWeb/Bindings/WindowProxy.cpp#L77
- https://github.com/SerenityOS/serenity/blob/987927a596cb773aee0069666d650e3f26f9f3ac/Userland/Libraries/LibWeb/Bindings/WindowProxy.cpp#L236

Reply:
Having to call `impl_from` just to get to an instance of `WindowObject` is a bit weird, is there a better way?

Reply:
That's how it currently works, but this will all change very soon when the wrappers go away.